labcontrol/db/process.py

```diff
@@ -2998,6 +2998,95 @@ def _format_sample_sheet_comments(principal_investigator=None,
 
         return ''.join(comments)
 
+    @staticmethod
+    def _set_control_values_to_plate_value(input_df, plate_col_name,
+                                           projname_col_name):
+        """ Update project name for control samples
+
+        Ensure that each sample plate included in the dataframe does not
+        contain experimental samples with more than (or less than) one
+        value in the column named projname_col_name. Assuming this is true, set
+        the project column value for each non-experimental sample to the value
+        of the project name for the (single) project on the non-experimental
+        sample's plate.
+
+        Parameters
+        ----------
+        input_df: pandas.DataFrame
+            A dataframe containing (at least) a column of plate names (having
+            the column name given in plate_col_name) and a column of project
+            names (having the column name given in projname_col_name)--e.g.,
+            Project Name on prep sheet or sample_proj_name on sample sheet--
+            and one row for each sample (both experimental and non-
+            experimental).  The value in the project name column must be None
+            for control (blank/positive control/etc) samples.
+        plate_col_name: str
+            The name of the column in input_df that contains the name of the
+            plate on which a given sample lies.
+        projname_col_name: str
+            The name of the column in input_df that contains the name of the
+            project associated with the given sample.
+
+        Returns
+        -------
+        input_df: pandas.DataFrame
+            The input dataframe, modified so that the controls have the same
+            (single) project name as the experimental samples on their sample
+            plate.
+
+        Raises
+        ------
+        ValueError
+            If any plate contains experimental samples from more (or fewer)
+            than one project.
+        """
+
+        problem_plate_messages = []
+
+        # create a mask to define all the NON-control rows for this plate
+        non_controls_mask = input_df[projname_col_name].notnull()
```

Comment:
Yes, they would implicitly be treated as controls--and would be assigned to the same project as everything else on their plate, which is exactly what the lab wanted (e.g. in the case of the AGP samples that were not in qiita but were plated on an AGP plate and that were *previously* given a project name of "" instead of the AGP project name of the rest of the samples on the plate).